packages/instrumentation-browser-navigation/src/instrumentation.ts

```diff
@@ -0,0 +1,406 @@
+      
+      const sameBase = fromBase === toBase;
+      const hashesAreDifferent = fromHash !== toHash;
+      const notRemovingHash = toHash !== ''; // Only true if we're not removing the hash
+      
+      return sameBase && hashesAreDifferent && notRemovingHash;
+    }
+  }
+
+  private _determineSameDocument(
+    changeState?: string | null,
+    navigationEvent?: any,
+    fromUrl?: string,
+    toUrl?: string
+  ): boolean {
+    // For Navigation API events, use the sameDocument property if available
+    if (changeState === 'navigate' && navigationEvent?.destination?.sameDocument !== undefined) {
+      return navigationEvent.destination.sameDocument;
+    }
+    
+    // For other navigation types, determine based on URL comparison
+    if (fromUrl && toUrl) {
+      try {
+        const fromURL = new URL(fromUrl);
+        const toURL = new URL(toUrl);
+        // Same document if origin is the same (cross-origin navigations are always different documents)
+        // In SPAs, route changes via pushState/replaceState are same-document navigations
+        return fromURL.origin === toURL.origin;
+      } catch {
+        // Fallback: assume same document for relative URLs or parsing errors
+        return true;
+      }
+    }
+    
+    // Default: if we can't determine URLs, assume it's a same-document navigation
+    // This handles cases where URL comparison fails
+    return true;
+  }
+
+  /**
+   * Determines if navigation is a hash change based on URL comparison
+   * A hash change is true if the URLs are the same except for the hash part
+   */
+  private _determineHashChange(
+    changeState?: string | null,
+    navigationEvent?: any,
+    fromUrl?: string,
+    toUrl?: string
+  ): boolean {
+    // For Navigation API events, use the hashChange property if available
+    if (changeState === 'navigate' && navigationEvent?.hashChange !== undefined) {
+      return navigationEvent.hashChange;
+    }
+    
+    // For all other cases, determine based on URL comparison
+    if (fromUrl && toUrl) {
+      return this._isHashChange(fromUrl, toUrl);
+    }
+    
+    return false;
+  }
+
+  /**
+   * Sanitizes URL according to OpenTelemetry specification:
+   * - Redacts credentials (username:password)
+   * - Redacts sensitive query parameters
+   * - Preserves fragment when available
+   */
+  private _sanitizeUrl(url: string): string {
+    try {
+      const urlObj = new URL(url);
+      
+      // Redact credentials if present
+      if (urlObj.username || urlObj.password) {
+        urlObj.username = 'REDACTED';
+        urlObj.password = 'REDACTED';
+      }
+      
+      // Redact sensitive query parameters
+      const sensitiveParams = [
+        'password', 'passwd', 'secret', 'api_key', 'apikey', 'auth', 'authorization',
+        'token', 'access_token', 'refresh_token', 'jwt', 'session', 'sessionid',
+        'key', 'private_key', 'client_secret', 'client_id', 'signature', 'hash'
+      ];
```

Comment:
Looks like this can be hoisted outside the `try/catch` to reduce the redundancy.